controllers/pipeline/pipeline_adapter.go

```diff
@@ -278,6 +301,23 @@ func (a *Adapter) getImagePullSpecFromSnapshotComponent(snapshot *applicationapi
 	return "", fmt.Errorf("couldn't find the requested component info in the given Snapshot")
 }
 
+// getComponentSourceFromSnapshotComponent gets the component source from the given Snapshot Component,
+func (a *Adapter) getComponentSourceFromSnapshotComponent(snapshot *applicationapiv1alpha1.Snapshot, component *applicationapiv1alpha1.Component) (*applicationapiv1alpha1.ComponentSource, error) {
+	for _, snapshotComponent := range snapshot.Spec.Components {
+		if snapshotComponent.Name == component.Name {
+			return &snapshotComponent.Source, nil
+		}
+	}
+	return nil, fmt.Errorf("couldn't find the requested component source info in the given Snapshot")
+}
+
+// getComponentSourceWithLastBuildCommit gets the component source from the given Component with LastBuiltCommit as Revision
+func (a *Adapter) getComponentSourceWithLastBuildCommit(component *applicationapiv1alpha1.Component) *applicationapiv1alpha1.ComponentSource {
+	componentSource := component.Spec.Source.DeepCopy()
+	componentSource.GitSource.Revision = component.Status.LastBuiltCommit
```

Comment:
There is a possibility that the `component.Status` doesn't have the `LastBuiltCommit` defined. I think we need to handle that case.

Reply:
Thanks. Add code to handle the missing `LastBuiltCommit`.

Reply:
Thanks, I made a follow-up comment on the updated code.